### PR TITLE
[IMP] website_forum: Change profile picture for forum users.

### PR DESCRIPTION
**Pull request by dpa-odoo**
Task : https://www.odoo.com/web#id=13410&view_type=form&model=project.task&menu_id=3942&action=327

Pad : https://pad.odoo.com/p/r.2sQxHcQI2aOmyBjp

### CODE REVIEW

File: addons/website_forum/controllers/main.py [hunk 1]
```diff
@@ -560,9 +561,22 @@ def open_partner(self, forum, partner_id=0, **post):
                 return werkzeug.utils.redirect("/forum/%s/user/%d" % (slug(forum), partner.user_ids[0].id))
         return werkzeug.utils.redirect("/forum/%s" % slug(forum))
 
+    @http.route(['/forum/user/<int:user_id>/remove_avatar'], type='http', auth="public", website=True)
+    def remove_avatar(self, user_id=0, **post):
+        user = request.env['res.users'].sudo().search([('id', '=', user_id)])
+        user.image = False
+        self.user_avatar(user_id)
```
Comment (rga-odoo) [inline addons/website_forum/controllers/main.py]: now anyone can remove my avatar

File: addons/website_forum/controllers/main.py [hunk 2]
```diff
@@ -560,9 +561,22 @@ def open_partner(self, forum, partner_id=0, **post):
...
+
     @http.route(['/forum/user/<int:user_id>/avatar'], type='http', auth="public", website=True)
     def user_avatar(self, user_id=0, **post):
         status, headers, content = binary_content(model='res.users', id=user_id, field='image', default_mimetype='image/png', env=request.env(user=openerp.SUPERUSER_ID))
+
+        user = request.env['res.users'].sudo().search([('id', '=', user_id)])
+        image_path = get_module_resource('base', 'static/src/img', 'avatar.png')
+        image = tools.image_resize_image_big(open(image_path, 'rb').read().encode('base64'))
+        if not user['image']:
+            user['image'] = image
```
Comment (rga-odoo) [inline addons/website_forum/controllers/main.py]: No, don't explicitly write avatar image.

File: addons/website_forum/controllers/main.py [hunk 3]
```diff
@@ -683,6 +697,8 @@ def edit_profile(self, forum, user, **kwargs):
 
     @http.route('/forum/<model("forum.forum"):forum>/user/<model("res.users"):user>/save', type='http', auth="user", methods=['POST'], website=True)
     def save_edited_profile(self, forum, user, **kwargs):
+
+        data = kwargs.get('ufile').read()
```
Comment (rga-odoo) [inline addons/website_forum/controllers/main.py]: rename data to image or file

File: addons/website_forum/static/src/js/website_forum.js [hunk 4]
```diff
@@ -28,6 +28,28 @@ if(!$('.website_forum').length) {
         }
     });
 
+    // Change image on website Edit Your Bio
+    $('#upload_file').change(function(){
+        if (this.files && this.files[0]) {
+            var reader = new FileReader();
+            reader.onload = function (e) {
+                $('#forum_image').attr('src', e.target.result);
```
Comment (rga-odoo) [inline addons/website_forum/static/src/js/website_forum.js]: don't use jqeury global selector

File: addons/website_forum/static/src/js/website_forum.js [hunk 5]
```diff
@@ -28,6 +28,28 @@ if(!$('.website_forum').length) {
...
+            }
+            reader.readAsDataURL(this.files[0]);
+        }
+    });
+
+    $('.oe_form_binary_file_clear').click(function(){
+        $.ajax({
+            url:'/forum/user/'+$("input[name='user_id']").val()+'/remove_avatar/'
+        }).then(function(status){
```
Comment (rga-odoo) [inline addons/website_forum/static/src/js/website_forum.js]: why ajax call ??

File: addons/website_forum/views/website_forum.xml [hunk 6]
```diff
@@ -1254,11 +1254,16 @@
         <h3>Edit Profile</h3>
         <div class="row">
             <div class="col-md-2">
-                <img class="img img-responsive img-circle" t-attf-src="/forum/user/#{user.id}/avatar"/>
+                <label for="upload_file">
+                    <i class="fa fa-pencil fa-1g pull-left oe_form_binary_file_edit" title="Edit"></i>
+                </label>
+                <i class="fa fa-trash-o fa-1g pull-right oe_form_binary_file_clear" title="Clear"></i>
+                <img id="forum_image" class="img img-responsive img-circle" t-attf-src="data:image/*;base64, #{user.image}"/>
```
Comment (rga-odoo) [inline addons/website_forum/views/website_forum.xml]: don't bind image data with HTML, use controller route or static path

File: addons/website_forum/static/src/js/website_forum.js (hunk 5 above)
Comment (rga-odoo) [inline addons/website_forum/static/src/js/website_forum.js]: can we clear hidden file input and post it with other data ??

File: addons/website_forum/controllers/main.py [hunk 7]
```diff
@@ -8,6 +8,7 @@
 from urllib2 import urlopen, URLError
 import base64
 
+from openerp.modules.module import get_module_resource
```
Comment (hmo-odoo) [inline addons/website_forum/controllers/main.py]: why import get_module_resource ?

File: addons/website_forum/controllers/main.py [hunk 8]
```diff
@@ -563,6 +564,7 @@ def open_partner(self, forum, partner_id=0, **post):
     @http.route(['/forum/user/<int:user_id>/avatar'], type='http', auth="public", website=True)
     def user_avatar(self, user_id=0, **post):
         status, headers, content = binary_content(model='res.users', id=user_id, field='image', default_mimetype='image/png', env=request.env(user=openerp.SUPERUSER_ID))
+
```
Comment (hmo-odoo) [inline addons/website_forum/controllers/main.py]: remove unnecessary diff

File: addons/website_forum/controllers/main.py [hunk 9]
```diff
@@ -565,6 +564,8 @@ def user_avatar(self, user_id=0, **post):
         status, headers, content = binary_content(model='res.users', id=user_id, field='image', default_mimetype='image/png', env=request.env(user=openerp.SUPERUSER_ID))
         if status == 304:
             return werkzeug.wrappers.Response(status=304)
+        if not content:
+            return request.redirect('/base/static/src/img/avatar.png')
```
Comment (msh-odoo) [inline addons/website_forum/controllers/main.py]: request.redirect ?

File: addons/website_forum/static/src/js/website_forum.js [hunk 10]
```diff
@@ -28,6 +28,33 @@ if(!$('.website_forum').length) {
         }
     });
 
+    $('.o_forum_profile_pic_edit').on('click', function(ev) {
+        ev.preventDefault();
+        $(this).closest('form').find('.o_forum_file_upload').trigger('click');
+    });
+
+    $('.o_forum_file_upload').on('change', function() {
+        if (this.files && this.files[0]) {
```
Comment (msh-odoo) [inline addons/website_forum/static/src/js/website_forum.js]: only this.files.length should work

**Comment by eLBati** [comment]
Hello @dpa-odoo see PR for 9.0: https://github.com/odoo/odoo/pull/9890
